packages/editor/src/components/inserter/menu.js

```diff
@@ -48,10 +48,12 @@ const MAX_SUGGESTED_ITEMS = 9;
 export const searchItems = ( items, searchTerm ) => {
 	const normalizedSearchTerm = normalizeTerm( searchTerm );
 	const matchSearch = ( string ) => normalizeTerm( string ).indexOf( normalizedSearchTerm ) !== -1;
+	const categories = getCategories();
```

Comment:
Could we retrieve the categories in the withSelect using select( 'core/blocks' ).getCategories() and then pass them to this function?
I think data retrieval functions of wp.blocks only exist because of historical reasons and they may be removed some day.
Maybe we can also add a test case to the searchItems tests matching using the categories.

Reply:
> Could we retrieve the categories in the withSelect using select( 'core/blocks' ).getCategories() and then pass them to this function?

Yes, we could, but I intentionally didn't do it - there are several more usages of `getCategories()` in this file (and not only in this file), so I think rewriting to use `withSelect` would be material for another PR.

> I think data retrieval functions of wp.blocks only exist because of historical reasons and they may be removed some day.

I realize this might be the case, but they don't even seem to be deprecated yet, so in that case I'd prefer staying consistent and following the example of code that's already merged and has proven to work well.

> Maybe we can also add a test case to the searchItems tests matching using the categories.

Good call! I've added a unit test case in https://github.com/WordPress/gutenberg/pull/9999/commits/11d1b697845a56cbb2b9b85426929a03efdad0a5